### PR TITLE
Code optimizations reported by static code analysis (Step 3) 

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
# What was changed:

- pass function parameters by const references instead of values;
- moved some variable declaration to lower level scope;
- simplified string initialization;
- fixed usage of invalid enum type;
- normalized variable and return types (int/bool).

***

Also includes changes from #24013 and #24023.

### CODE REVIEW

File: src/advanced_inv.cpp [hunk 1]
```diff
@@ -1417,10 +1417,10 @@ void advanced_inventory::display()
             do_return_entry();
             return;
         }
-        dest = ( src == left ? right : left );
+        dest = ( src == advanced_inventory::side::left ? advanced_inventory::side::right : advanced_inventory::side::left );
```
Comment (ZhilkinSerg) [inline src/advanced_inv.cpp]: Invalid enum type was used here.

**Comment by kevingranade** [comment]
Needs a rebase or merge, the diff shows the union of this PR and the previous one.

This wouldn't be a critical problem, except I tend to review this kind of PR on a browser when I'm away from my PC, so I can't use git commands to get a clean diff myself.

**Comment by ZhilkinSerg** [comment]
Time to learn `rebase` then :)

File: src/iexamine.cpp [hunk 2]
```diff
@@ -2201,9 +2201,9 @@ void iexamine::fvat_full( player &p, const tripoint &examp )
         return;
     }
 
-    const std::string booze_name = g->m.i_at( examp ).front().tname();
```
Comment (kevingranade) [inline src/iexamine.cpp]: This is capturing the name, the next line removes the item.

File: src/item.cpp [hunk 3]
```diff
@@ -5531,9 +5531,6 @@ bool item::detonate( const tripoint &p, std::vector<item> &drops )
     } else if( type->ammo && ( type->ammo->special_cookoff || type->ammo->cookoff ) ) {
         long charges_remaining = charges;
         const long rounds_exploded = rng( 1, charges_remaining );
-        // Yank the exploding item off the map for the duration of the explosion
-        // so it doesn't blow itself up.
-        item temp_item = *this;
```
Comment (kevingranade) [inline src/item.cpp]: I'm not sure what this is doing, but this code has broken a lot in the past, please remove this and put the change it it's own PR.

File: src/npcmove.cpp [hunk 4]
```diff
@@ -3279,8 +3279,8 @@ void npc::do_reload( item &it )
                                        it.ammo_capacity() - it.ammo_remaining() ) );
     int reload_time = item_reload_cost( it, *usable_ammo, qty );
     // @todo: Consider printing this info to player too
-    const std::string ammo_name = usable_ammo->tname();
```
Comment (kevingranade) [inline src/npcmove.cpp]: This is a bug, usable_ammo is no longer valid for deference after the next line.

File: src/game.cpp [hunk 5]
```diff
@@ -3047,9 +3047,9 @@ bool game::handle_action()
             break;
 
         case ACTION_FIRE_BURST: {
-            gun_mode_id mode = u.weapon.gun_get_mode_id();
```
[the remark this answers is not in the document]
Comment (ZhilkinSerg) [reply src/game.cpp]: I will rename it to original mode.

[the remark this answers is not in the document]
Comment (kevingranade) [reply src/game.cpp]: Perfect renaming it to make it clearer what doing.
